crates/bevy_core_widgets/src/core_radio.rs

```diff
@@ -0,0 +1,206 @@
+use accesskit::Role;
+use bevy_a11y::AccessibilityNode;
+use bevy_app::{App, Plugin};
+use bevy_ecs::hierarchy::{ChildOf, Children};
+use bevy_ecs::query::Has;
+use bevy_ecs::system::In;
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    observer::On,
+    query::With,
+    system::{Commands, Query, SystemId},
+};
+use bevy_input::keyboard::{KeyCode, KeyboardInput};
+use bevy_input::ButtonState;
+use bevy_input_focus::FocusedInput;
+use bevy_picking::events::{Click, Pointer};
+use bevy_ui::{Checked, InteractionDisabled};
+
+/// Headless widget implementation for a "radio button group". This component is used to group
+/// multiple [`CoreRadio`] components together, allowing them to behave as a single unit. It
+/// implements the tab navigation logic and keyboard shortcuts for radio buttons.
+///
+/// The [`CoreRadioGroup`] component does not have any state itself, and makes no assumptions about
+/// what, if any, value is associated with each radio button. Instead, the "value" of the group is
+/// the entity id of the selected button. The app can then derive the selected value from this using
+/// app-specific data.
+#[derive(Component, Debug)]
+#[require(AccessibilityNode(accesskit::Node::new(Role::RadioGroup)))]
+pub struct CoreRadioGroup {
+    /// Callback which is called when the selected radio button changes.
+    pub on_change: Option<SystemId<In<Entity>>>,
+}
+
+/// Headless widget implementation for radio buttons. These should be enclosed within a
+/// [`CoreRadioGroup`] widget, which is responsible for the mutual exclusion logic.
+///
+/// According to the WAI-ARIA best practices document, radio buttons should not be focusable,
+/// but rather the enclosing group should be focusable.
+/// See <https://www.w3.org/WAI/ARIA/apg/patterns/radio>/
+#[derive(Component, Debug)]
+#[require(AccessibilityNode(accesskit::Node::new(Role::RadioButton)), Checked)]
+pub struct CoreRadio;
```

Comment:
```suggestion
pub struct CoreRadio(Option<Entity>);
```
Maybe to enforce exclusivity you could propagate down the entity id of the currently selected radio button and store it in each `CoreRadio` component. Then to check if a radio button is selected you'd compare it's own entity id against the id stored in its `CoreRadio` component, ensuring only one button in a radio group can be selected at any time.

Reply:
That could still result in multiple selections within a group if a selected radio button is reparented or something though, I suppose.

Reply:
Yeah I understand it's difficult without reactivity. It's just that mutual exclusivity is the defining characteristic of radio buttons, so expecting users to have to manage and enforce it themselves feels like it defeats the purpose of having this widget. 

Maybe it's just not possible or at least not worth the effort at the moment though and this is the best we can do.